Do not attempt to compile unwind.cpp on aarch64

malfet opened the pull request:
Summary:
As almost entire unwinding logic is build around x86_64 ABI
In essence, this reverts https://github.com/pytorch/pytorch/pull/104707 and adds `#ifndef FBCODE_CAFFE2` guards around `symbolize` dummy

Use nested namespaces, as PyTorch is finally C++ compatible.
Remove extraneous semicolon spotted by clang-tidy.

Fixes https://github.com/pytorch/pytorch/issues/113208

Test Plan: CI + `buck2 build fbcode//mode/opt fbcode//caffe2/torch/fb/model_transform/fx2trt/packaging:generate_merge_net_file -c fbcode.arch=aarch64`

Differential Revision: D51358469

aaronenyeshi reviewed (approved):
LGTM, please check lintrunner issues

malfet commented:
/easycla

malfet committed:
Do not attempt to compile unwind.cpp on aarch64 (#113782)

Summary:

As almost entire unwinding logic is build around x86_64 ABI
In essence, this reverts https://github.com/pytorch/pytorch/pull/104707 and adds `#ifndef FBCODE_CAFFE2` guards around `symbolize` dummy

Use nested namespaces, as PyTorch is finally C++ compatible.
Remove extraneous semicolon spotted by clang-tidy.

Fixes https://github.com/pytorch/pytorch/issues/113208

Test Plan: CI + `buck2 build fbcode//mode/opt fbcode//caffe2/torch/fb/model_transform/fx2trt/packaging:generate_merge_net_file -c fbcode.arch=aarch64`

Reviewed By: aaronenyeshi

Differential Revision: D51358469

malfet commented:
@pytorchbot merge -f "I've run tests twice already, only linter was red"
